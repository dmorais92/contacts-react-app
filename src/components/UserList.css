.User .row {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
}
.row .logo {
  height: 10vh;
}
.row .main-info {
  display: flex;
  flex-flow: column wrap;
}
.main-info h3 {
  margin: 0;
}
