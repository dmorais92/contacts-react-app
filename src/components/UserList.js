import React, { PureComponent, Fragment } from "react";
import PropTypes from "prop-types";
import User from "./User";

class UserList extends PureComponent {
  static propTypes = {
    users: PropTypes.array,
  };

  componentDidMount() {
    if (this.props.getUsers) {
      this.props.getUsers();
    }
  }

  render() {
    const {users, isFetchingUsers} = this.props;

    return (
      <Fragment>
        {
          users && users.length ?
            users.map(user => <User user={user} key={user.name}/>)
          :
            <h3>{isFetchingUsers ? 'Loading contacts...' : 'No contacts found'}</h3>
        }
      </Fragment>
    );
  }
}

export default UserList;
