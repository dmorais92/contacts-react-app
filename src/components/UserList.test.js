import React from 'react';
import ReactDOM from 'react-dom';
import UserList from './UserList';
import renderer from 'react-test-renderer';
import TEST_USERS from '../dev/users.json';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders a list of users', () => {
  const component = renderer.create(
    <UserList users={TEST_USERS}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
