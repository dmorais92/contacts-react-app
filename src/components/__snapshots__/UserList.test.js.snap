// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of users 1`] = `
Array [
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Leanne Graham
        </h3>
        <span>
          Sincere@april.biz
        </span>
        <span>
          Bret
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Ervin Howell
        </h3>
        <span>
          Shanna@melissa.tv
        </span>
        <span>
          Antonette
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Clementine Bauch
        </h3>
        <span>
          Nathan@yesenia.net
        </span>
        <span>
          Samantha
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Patricia Lebsack
        </h3>
        <span>
          Julianne.OConner@kory.org
        </span>
        <span>
          Karianne
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Chelsey Dietrich
        </h3>
        <span>
          Lucio_Hettinger@annie.ca
        </span>
        <span>
          Kamren
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Mrs. Dennis Schulist
        </h3>
        <span>
          Karley_Dach@jasper.info
        </span>
        <span>
          Leopoldo_Corkery
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Kurtis Weissnat
        </h3>
        <span>
          Telly.Hoeger@billy.biz
        </span>
        <span>
          Elwyn.Skiles
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Nicholas Runolfsdottir V
        </h3>
        <span>
          Sherwood@rosamond.me
        </span>
        <span>
          Maxime_Nienow
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Glenna Reichert
        </h3>
        <span>
          Chaim_McDermott@dana.io
        </span>
        <span>
          Delphine
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
  <div
    className="Card vertical User"
  >
    <div
      className="row"
    >
      <i
        className="Icon far fa-user m primary"
      />
      <div
        className="main-info"
      >
        <h3>
          Clementina DuBuque
        </h3>
        <span>
          Rey.Padberg@karina.biz
        </span>
        <span>
          Moriah.Stanton
        </span>
      </div>
      <i
        className="Icon fas fa-angle-down m primary"
        onClick={[Function]}
      />
    </div>
    
  </div>,
]
`;
